src/boot/efi/linux.c

```diff
@@ -15,8 +15,48 @@
 #include "linux.h"
 #include "missing_efi.h"
 #include "pe.h"
+#include "secure-boot.h"
 #include "util.h"
 
+static EFI_SECURITY_FILE_AUTHENTICATION_STATE orig_auth;
+static EFI_SECURITY2_FILE_AUTHENTICATION orig_auth2;
+static const void *trusted_buffer;
+static const EFI_DEVICE_PATH *trusted_device_path;
```

Comment:
hmm, static vars are ugly. I guess we have no other nice way to pass this around, but could we make this a single pointer only?

i.e. move the four fields into a struct, then allocate that struct on the stack as usual, and only have a single static pointer? That static pointer would be the only thing that is static and the only thing that tells us if the redirection is in effect or not, depending on if it is NULL or not.